filter.py

```diff
@@ -1,51 +1,54 @@
-def filter(data, state, label, start_date, end_date, datetype):
+def filter_releases(data, state, label, start_date, end_date, datetype):
+  """Filters by all of the criteria shown below, returns an array of filtered releases."""  # pylint: disable=line-too-long
+  # convert variables from unix type to something usable
+  state = int(state)
+  start_date = int(start_date)
+  end_date = int(end_date)
 
-    # convert variables from unix type to something usable
-    state = int(state)
-    start_date = int(start_date)
-    end_date = int(end_date)
+  filtered = []
 
-    filtered = []
+  # Validate end_date value
+  if end_date <= 0:
+    end_date = (datetime.datetime.now()-datetime.datetime(1970, 1, 1)).total_seconds()
 
-    # Validate end_date value
-    if end_date <= 0:
-        end_date = (datetime.datetime.now()-datetime.datetime(1970,1,1)).total_seconds()
+  for item in data.values():
+    if item[datetype] >= start_date and item[datetype] <= end_date:
+      if item['state'] == state or state == 0:
+        if label != 'null':
+          for l in item['labels']:
+            if l == label:
+              filtered.append(item)
+        else:
+          filtered.append(item)
 
-    for item in data.items():
-        if item[1][datetype] >= start_date and item[1][datetype] <= end_date:
-            if item[1]['state'] == state or state == 0:
-                if label != 'null':
-                    for l in item[1]['labels']:
-                        if l == label:
-                            filtered.append(item[1])
-                else:
-                    filtered.append(item[1])
+  return filtered
 
-    return filtered
 
-
-# sorts 'unsorted' according to 'sort_method'. See https://docs.google.com/document/d/1JDD_NX2XVL7yqYcfFOqkef1FKv98nrlRFJ0OpSZprMU/ for enumerations
 def sort(unsorted, sort_method):
-    sort_method = int(sort_method)
-    if sort_method == 1:
-        result = sorted(unsorted, key=lambda k: k['name'], reverse=True)
-    elif sort_method == 2:
-        result = sorted(unsorted, key=lambda k: k['name'])
-    elif sort_method == 3:
-        result = sorted(unsorted, key=lambda k: k['started'], reverse=True)
-    elif sort_method == 4:
-        result = sorted(unsorted, key=lambda k: k['started'])
-    elif sort_method == 5:
-        result = sorted(unsorted, key=lambda k: k['last_modified'], reverse=True)
-    elif sort_method == 6:
-        result = sorted(unsorted, key=lambda k: k['last_modified'])
-    elif sort_method == 7:
-        result = sorted(unsorted, key=lambda k: k['last_active_task'], reverse=True)
-    elif sort_method == 8:
-        result = sorted(unsorted, key=lambda k: k['last_active_task'])
-    return result
+  """Sorts 'unsorted' according to 'sort_method'."""
+
+  """See
+  https://docs.google.com/document/d/1JDD_NX2XVL7yqYcfFOqkef1FKv98nrlRFJ0OpSZprMU/"""  # pylint: disable=pointless-string-statement
+
+  sort_method = int(sort_method)
```

Comment:
Same comment here. Please use an enum and document in code. Reader should not have to open that doc to figure how this is used.

Reply:
also is there a way to simplify this API ? Can't you use something like:
https://specs.openstack.org/openstack/api-wg/guidelines/pagination_filter_sort.html

Allowing sorting on multiple keys.

Reply:
should we define the enum classes in this file or as a separate module?

Reply:
could you explain how you would want sorting on multiple keys to appear on the UI

Reply:
You should be defining the classes in release_data_abc.py.

The API might allow sorting on multiple fields, but the UI might impose the sorting based on the main key. For example if you are sorting by name, you might impose the second sort to be by last_modified. If you are sorting by last_modified the second sort would be by name. That way you get consistent results.